docs/source/joblib.rst

```diff
@@ -52,6 +52,15 @@ validated parameter search as follows.
    with parallel_backend('dask.distributed', scheduler_host='localhost:8786'):
        search.fit(digits.data, digits.target)
 
+## Nested Parallelism
+
+Note that this code will not work if you distribute it via dask in order to implement nested parallelism. The reason is that each worker will have a fresh copy of sklearn and `from sklearn.externals.joblib import parallel_backend` will not apply to them, resulting in a "KeyError" when trying to distribute, but no problem when running the inner loop locally. The solution is 
+
+.. code-block:: python
+
+    from distributed.joblib import DistributedBackend 
+    register_parallel_backend('distributed', DistributedBackend, make_default=True)
```

Comment:
Lets say that your machine had four cores.  `Client()` would create a local cluster with four workers.  Four of your original `function` calls would start and would themselves launch other tasks on which they would wait.  However none of these new tasks would be able to start, because all of the workers are busy with the original calls to `function`.

The solution here is for tasks that are mostly doing dask-things (waiting on futures) to leave the threadpool and make way for more tasks to participate.  If you have a chance you might want to look over this [documentation PR](https://github.com/dask/dask/pull/2501) and in particular look for the term `secede` (which currently depends on https://github.com/dask/distributed/pull/1201).